### DISCUSSION TITLE
Elastic search Missing, salt errors after highstate, after update

### DISCUSSION THREAD

**Discussion by ghost** [Security-Onion-Solutions/securityonion#5666] [Unsupported Versions] [answered]
### Discussed in https://github.com/Security-Onion-Solutions/securityonion/discussions/5476

https://github.com/Security-Onion-Solutions/securityonion/discussions/5476#discussioncomment-1381250

**Comment by ghost** [comment]
On distributed Setting, on Search Node.
Graphana Status Search Node, Process Status Fault
so-status -> so-elasticsearch -> Missing
After doing a salt .\ state.highstate
so-status -> so-elasticsearch -> Pending

From Master:
sudo salt .\ state.highstate completes with 7 task failures on Search Node with several errors:

wazuhdir -> false -> FileExistsError: [Errno 17] File exists: '/nsm'
registertheagent -> false -> ERROR: Wazuh manager did not become ready after 30 attempts; unable to proceed with registration)
nsmdir -> false -> [Errno 17] File exists: '/nsm'
so-elasticsearch -> docker_container_running -> False -> 'Error 500: error while creating mount source path '/nsm/elasticsearch': mkdir /nsm: file exists'.
importdir -> false -> [Errno 17] File exists: '/nsm'
nsmlsdir -> false -> [Errno 17] File exists: '/nsm'
run_module_setup -> false -> One or more requisite failed: elasticsearch.so-elasticsearch

After state-highstate completes, so-status -> so-elasticsearch stattus goes back to Missing

No recent information on /opt/so/log/elasticsearch/.log (-7 days, log ends on Sept 14)

Thanks,

Edgar

**Comment by dougburks** [comment, marked as answer]
Were there any errors when you updated?

What is the output of `sudo docker images`?

Have you tried running `sudo so-docker-refresh`?

**Reply by ghost** [reply]
Hello Doug,
No errors during the update, the errors only come to light during the salt highstate.
Q: is there a way to list only the failed actions after the salt?

the output of sudo docker images, from the search node:

REPOSITORY                                                       TAG       IMAGE                                                                                                                ID       CREATED        SIZE
ManagerNode:5000/security-onion-solutions/so-telegraf        2.3.70    311de                                                                                                               e6c9cab   5 weeks ago    91.5MB
ManagerNode:5000/security-onion-solutions/so-soc             2.3.70    1553e                                                                                                               77f29b5   6 weeks ago    85.4MB
ManagerNode:5000/security-onion-solutions/so-nginx           2.3.70    c8ee1                                                                                                               7546d79   7 weeks ago    87.4MB
MangerNode:5000/security-onion-solutions/so-logstash        2.3.70    afdc4                                                                                                               5ce9994   7 weeks ago    1.62GB
ManagerNode:5000/security-onion-solutions/so-curator         2.3.70    e8ea5                                                                                                               ec5b542   7 weeks ago    123MB
ManagerNode:5000/security-onion-solutions/so-wazuh           2.3.70    a9e25                                                                                                               6f9c662   7 weeks ago    884MB
ManagerNode:5000/security-onion-solutions/so-elasticsearch   2.3.70    7b6d0                                                                                                               a4b011e   7 weeks ago    1.57GB
ManagerNode:5000/security-onion-solutions/so-filebeat        2.3.70    23115                                                                                                               5ee4a4b   7 weeks ago    628MB
ManagerNode:5000/security-onion-solutions/so-soc             2.3.61    ba6a8                                                                                                               e2083c8   2 months ago   85.3MB
ManagerNode:5000/security-onion-solutions/so-logstash        2.3.61    4599f                                                                                                               ec6fecc   2 months ago   1.62GB
ManagerNode:5000/security-onion-solutions/so-elasticsearch   2.3.61    43697                                                                                                               51819e8   2 months ago   1.57GB
ManagerNode:5000/security-onion-solutions/so-filebeat        2.3.61    8d82b                                                                                                               549386c   2 months ago   628MB
ManagerNode:5000/security-onion-solutions/so-wazuh           2.3.61    10f29                                                                                                               b472b6c   2 months ago   884MB
ManagerNode:5000/security-onion-solutions/so-nginx           2.3.61    ebf75                                                                                                               91b8bea   2 months ago   87.4MB
ManagerNode:5000/security-onion-solutions/so-curator         2.3.61    37382                                                                                                               6a118fd   2 months ago   123MB
ManagerNode:5000/security-onion-solutions/so-telegraf        2.3.61    556c1                                                                                                               409a918   2 months ago   91.5MB

The output of so-docker-refresh from Manager, ends correctly, many "Layer already exists"

On the Search node, after docker refresh:
so-elasticsearch-status -> Missing
so-elasticsearch-restart ->
Error response from daemon: No such container: so-elasticsearch
Error: No such container: so-elasticsearch
----------
          ID: nsmesdir
    Function: file.directory
        Name: /nsm/elasticsearch
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python3.6/site-packages/salt/state.py", line 2172, in call
                  *cdata["args"], **cdata["kwargs"]
                File "/usr/lib/python3.6/site-packages/salt/loader.py", line 1235, in __call__
                  return self.loader.run(run_func, *args, **kwargs)
                File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2268, in run
                  return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
                File "/usr/lib/python3.6/site-packages/contextvars/__init__.py", line 38, in run
                  return callable(*args, **kwargs)
                File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2283, in _run_as
                  return _func_or_method(*args, **kwargs)
                File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2316, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python3.6/site-packages/salt/states/file.py", line 3658, in directory
                  win_inheritance=win_inheritance,
                File "/usr/lib/python3.6/site-packages/salt/states/file.py", line 1315, in _makedirs
                  path=name, user=user, group=group, mode=dir_mode
                File "/usr/lib/python3.6/site-packages/salt/loader.py", line 1235, in __call__
                  return self.loader.run(run_func, *args, **kwargs)
                File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2268, in run
                  return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
                File "/usr/lib/python3.6/site-packages/contextvars/__init__.py", line 38, in run
                  return callable(*args, **kwargs)
                File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2283, in _run_as
                  return _func_or_method(*args, **kwargs)
                File "/usr/lib/python3.6/site-packages/salt/modules/file.py", line 6384, in makedirs_
                  mkdir(directory_to_create, user=user, group=group, mode=mode)
                File "/usr/lib/python3.6/site-packages/salt/modules/file.py", line 6321, in mkdir
                  makedirs_perms(directory, user, group, mode)
                File "/usr/lib/python3.6/site-packages/salt/modules/file.py", line 6413, in makedirs_perms
                  os.mkdir(name)
              FileExistsError: [Errno 17] File exists: '/nsm'
     Started: 14:09:06.009671
    Duration: 11.65 ms
     Changes:
----------
          ID: so-elasticsearch
    Function: docker_container.running
      Result: False
     Comment: Created container 'so-elasticsearch'. Failed to start container 'so-elasticsearch': 'Error 500: error while creating mount source path '/nsm/elasticsearch': mkdir /nsm: file exists'.
     Started: 14:09:06.179080
    Duration: 1414.515 ms
     Changes:
----------

so-elasticsearch-status -> pending

**Reply by dougburks** [reply]
That's strange. I don't believe we've had any other reports like this. Is there anything special about /nsm or /nsm/elasticsearch, like being mounting to a different filesystem or something like that?

**Reply by ghost** [reply]
Hello Doug,
No. .nsm is local to the server, its a physical system, and its own RAID Volume Group.

sdb                               8:16   0   5.5T  0 disk
└─sdb1                            8:17   0   5.5T  0 part
  └─vg--nsm-lv--nsm             253:3    0   5.5T  0 lvm  /nsm

As far as I can guess, the update simply sees /NSM as already occupied by elastic search data, and wont use it.

Thanks,

Edgar

**Reply by ghost** [reply]
I think I just found it ....

 ls -latr /
ls: cannot access /nsm: Input/output error
d??????????   ? ?    ?        ?            ? nsm

The filesystem is gone form nsm????

**Reply by ghost** [reply]
So, I think we can close this one. 
Its a local filesystem issue.
Thanks for the extra set of eyes.

Edgar

**Comment by m0duspwnens** [comment]
@edgarmtoro Can you verify that `/nsm` is a directory and not a file?
